src/rust/ensogl/src/math/geometry/triangle.rs

```diff
@@ -0,0 +1,128 @@
+//! This module defines an computations related to triangle geometry.
+
+use crate::math::algebra::{Acos,Sin,Cos,Sqrt};
+use crate::math::algebra::Field;
+
+use core::f32::consts::PI;
+
+/// Represents a triangle through its angles and side lengths. This struct is only meant to be a
+/// return type within this module and should not otherwise be instantiated.
+///
+///
+/// Triangle Schematic
+/// ----------------------
+///
+///                           C
+///                          / \
+///                        /    -\
+///                      -/ gamma --\
+///                   --/            --\
+///               b -/                  --\ a
+///              --/                       --\
+///            -/                             --\
+///          -/                                  --\
+///       --/                                       --\
+///     -/  alpha                                   beta-\
+///   -/--------------------------------------------------- B
+///  A                              c
+///
+/// Where a, b and c are the length of the respective side.
+/// Where alpha, beta and gamma are the respective angle.
+///
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct Triangle<S> {
+    pub side_length_a   : S,
+    pub side_length_b   : S,
+    pub side_length_c   : S,
+    pub angle_alpha_rad : S,
+    pub angle_beta_rad  : S,
+    pub angle_gamma_rad : S,
+    // Prevent instantiation as something other than a result type from this module.
+    dummy: (),
```

Comment:
Oh no, dont do it. If you want to prevent initialization just don't make fields public.

Reply:
Replaced it with a struct with private fields and getters.